server/server.go

```diff
@@ -166,24 +169,42 @@ func Main(keyP crypki.KeyIDProcessor) {
 	recoveryHandler := func(p interface{}) (err error) {
 		return status.Errorf(codes.Internal, "internal server error")
 	}
+
+	var server *http.Server
+	interceptors := []grpc.UnaryServerInterceptor{
+		recovery.UnaryServerInterceptor(recovery.WithRecoveryHandler(recoveryHandler)),
+	}
+	if cfg.ShutdownOnFrequentSigningFailure {
+		criteria := cfg.ShutdownOnSigningFailureCriteria
+		shutdownCounterConfig := shutdownCounterConfig{
+			consecutiveCountLimit: int32(criteria.ConsecutiveCountLimit),
+			timeRangeCountLimit:   int32(criteria.TimerCountLimit),
+			tickerDuration:        time.Duration(criteria.TimerDurationSecond) * time.Second,
+			shutdownFn: func() {
```

Comment:
can we add a log stating the server is going to be shutdown due to this? Would be useful to differentiate this vs server crashing.